libs/langchain/langchain/document_loaders/parsers/audio.py

```diff
@@ -298,3 +299,189 @@ def lazy_parse(self, blob: Blob) -> Iterator[Document]:
                 page_content=res.normalized_text,
                 metadata={"source": blob.source},
             )
+
+
+class AzureSpeechServiceParser(BaseBlobParser):
+    """Loads an Audio with azure.cognitiveservices.speech."""
+
+    def __init__(self, **kwargs: Any) -> None:
```

Comment:
1) Could you add a link to the relevant pages on azure that explain the API?
2) Could you expand the kwargs and document the variable meanings?
3) Could we add a parameter to control the polling interval (i.e., sleep is hard-coded to 0.5 seconds right now, we should expose this to the user)

---

nit: It's common to use logger rather than print statements in production code, you could consider swapping to using logger.info / logger.error as appropriate. (not a requirement to merge since we have some other parsers that are using print right now)

Reply:
Dear eyurtsev,

I fixed all the issues in the commit [7b8372d](https://github.com/langchain-ai/langchain/pull/13988/commits/7b8372db69bfb905459400218afc71f0948a9d91):

I Added Azure Speech service official documents and code pages.
I expanded and documented the kwargs meanings.
I added a parameter to control the transcribe job's polling interval.

Furthermore, I did all the unit tests in the test_audio.py file and here's the evidence:
![Screenshot 2023-12-03 at 03 26 35](https://github.com/langchain-ai/langchain/assets/18718740/83eeb0b6-7bc9-4b14-8acc-3b009e5468f9)

 Look forward to seeing feedback from you. Thanks.